File: trainsim-client/src/models/Leg.ts
import Place from "./Place";

export default class Leg {
    private readonly _id: string;
    private readonly _routeId?: string | undefined;
    private readonly _distance: number;
    private readonly _places: readonly Place[];

    private constructor(id: string, routeId: string | undefined, distance: number,
        places: readonly Place[]) {
        this._id = id;
        this._routeId = routeId;
        this._distance = distance;
        this._places = places
    }

    public static walk(id: string, distance: number, places: readonly Place[]) {
        return new Leg(id, undefined, distance, places);
    }

    public static transit(id: string, routeId: string, distance: number,
        places: readonly Place[]) {
        return new Leg(id, routeId, distance, places);
    }

    public get id(): string {
        return this._id;
    }

    public get isWalking(): boolean {
        return this._routeId === undefined;
    }

    public get routeId(): string | undefined {
        return this._routeId;
    }

    public get distance(): number {
        return this._distance;
    }

    public get places(): readonly Place[] {
        return this._places;
    }
}

File: trainsim-itinerary-microservice/src/main/java/edu/drexel/trainsim/itinerary/models/Leg.java
package edu.drexel.trainsim.itinerary.models;

import java.util.List;
import java.util.UUID;

public class Leg {
    private final UUID id;
    private final String routeId;
    private final List<Place> places;
    private final double distance;

    public Leg(UUID id, String routeId, List<Place> places, double distance) {
        this.id = id;
        this.routeId = routeId;
        this.places = places;
        this.distance = distance;
    }

    public UUID getId() {
        return id;
    }

    public String getRouteId() {
        return routeId;
    }

    public List<Place> getPlaces() {
        return places;
    }

    public double getDistance() {
        return distance;
    }
}
